kernel/src/common/result.rs

```diff
@@ -0,0 +1,18 @@
+//! result.rs -- Standard return type for invoking operations, returning
+//! success or an error code.
+//!
+//!  Author: Philip Levis <pal@cs.stanford.edu>
+//!  Date: Nov 21, 2016
+
+enum Result {
+    SUCCESS,
+    FAIL, // Generic failure condition
+    EBUSY, // Underlying system is busy; retry
+    EALREADY, // The state requested is already set
```

Comment:
Is there a good example of a situation where you wouldn't want to simply return SUCCESS instead of EALREADY?